topics/data-science/tutorials/python-math/tutorial.md

```diff
@@ -0,0 +1,361 @@
+- Translate some known math functions (e.g. euclidean distance, root algorithm) into python to transfer concepts from mathematics lessons directly into Python.
+
+time_estimation:  30M
+key_points:
+- Converting mathematics equation to python is pretty easy!
+- In real life you'll occasionally need to do this, either converting from a formula you read in a paper, or a description of an algorithm, into code that you'll re-use.
+- Did you forget how a particular module or function works? Try `help()`
+
+enable: false
+subtopic: python-modular
+contributors:
+  - hexylena
+  - dirowa
+
+priority: 1
+notebook:
+  language: python
+---
+
+Here we'll learn some of the fundamentals of python and how to do basic maths in Python.
+
+> ### Agenda
+>
+> In this tutorial, we will cover:
+>
+> 1. TOC
+> {:toc}
+>
+{: .agenda}
+
+
+# Python Fundamentals
+
+## Variables
+
+Any Python interpreter can be used as a calculator. We can do simple sums
+
+```python
+3 + 5 - 4
+```
+
+Multiplication
+
+```python
+5 * 4
+```
+
+Division
+
+```python
+5 / 4
+```
+
+Just like you've probably learned in maths courses, we can assign values to variables
+
+```python
+x = 10
+y = 2.5
+z = 3
+```
+
+From now on, whenever we use `x` or `y` or `z`, Python will substitute the value we assigned to it.
+You'll notice when we assign things to a variable, nothing is printed out. If we've assigned a value to a variable we can print it by doing:
+
+```python
+print(x)
+print(y, z)
+```
+
+You can print out multiple variables if you separate them with a comma `,`.
+
+
+> ### {% icon tip %} Tip: Variable Names
+> In Python, variable names:
+>
+> - can include letters, digits, and underscores
+> - cannot start with a digit
+> - are case sensitive
+>
+> This means that, for example:
+>
+> - `x` is a valid name
+> - `weight0` is a valid variable name, whereas `0weight` is not
+> - `weight` and `Weight` are different variables
+{: .tip}
+
+```python
+x * (y + z)
+```
+
+What do you think the following will result in? The best way to find out is trying things for yourself. If it's wrong, can you correct it?
+
+```python
+xy
+```
+
+Time to check what we've learned!
+
+> ### {% icon question %} Exercise 0: Simple Equations
+> Given the equation:
+>
+> $$y = x * 921 + 534$$
+>
+> What is the value of `y` when `x = 452`
+>
+> > ### {% icon solution %} Solution
+> > ```
+> > x = 452
+> > y = x * 921 + 534
+> > # Remember to print it out!
+> > print(y)
+> > ```
+> >
+> > 416826
+> {: .solution}
+{: .question}
+
+```python
+# Test solutions here!
+# By the way, lines starting with a # are comment lines!
+# You can use that to take notes, and not affect how your code runes
+# People use it for documentation: explaining what the function does,
+# what a variable means, why they chose this or that algorithm.
+```
+
+## Functions
```

Comment:
@hexylena 
Should this be here ? 
Next part is all about Functions and this is a bit confusing here

Reply:
it's a good point. maybe that should move to `functions` and then have a more maths focused comparison here. Good suggestion